src/webview/SampleView.ts

```diff
@@ -135,14 +136,41 @@ export class SampleView extends View {
     }
 
     private async getSamplePage(options: any): Promise<any> {
-
+        let validInternetConnection: boolean = true;
+        const outputAdapter: VSCodeBlockchainOutputAdapter = VSCodeBlockchainOutputAdapter.instance();
         const templatePath: string = path.join(__dirname, '..', '..', '..', 'templates', 'SampleView.ejs');
 
         const converter: showdown.Converter = new showdown.Converter();
-        const response: any = await Axios.get(options.sample.readme);
-        const text: string = response.data;
-        const html: string = converter.makeHtml(text);
-        options.html = html;
+        try {
+            const response: any = await Axios.get(options.sample.readme);
+            const text: string = response.data;
+            const html: string = converter.makeHtml(text);
+            options.html = html;
+        } catch (error) {
+            validInternetConnection = false;
+        }
+
+        if (!validInternetConnection) {
+            const repositoryRegistry: RepositoryRegistry = RepositoryRegistry.instance();
+            const repoIsCloned: boolean = repositoryRegistry.exists(options.repositoryName);
+            if (repoIsCloned) {
+                const clonedRepo: RepositoryRegistryEntry = repositoryRegistry.get(options.repositoryName);
+                const repoPath: string = clonedRepo.path;
+                const readMeOnJSON: string = options.sample.readme;
+                const splitPath: string[] = readMeOnJSON.split(options.repositoryName);
+
+                const branchedPath: string = splitPath[1];
+                const endOfBranchName: number = branchedPath.indexOf('/', 1);
+                const slicedPath: string = branchedPath.slice(endOfBranchName + 1, branchedPath.length);
+
+                const readMe: string = await fs.readFileSync(path.join(repoPath, slicedPath), 'utf8');
```

Comment:
this should be `fs.readFile(.....` you don't need to use the synchronous version and use await

Reply:
Will change that. Thanks

Reply:
Fixed